# Update test utilities for FIPS compatibility:
# - Use JcaContentSignerBuilder instead of Bc*ContentSignerBuilder (FIPS-compatible)
# - Use SHA256withRSA instead of SHA1withRSA (SHA1 disallowed in FIPS 140-3)
# - Use AES_128_CBC instead of PBE_SHA1_3DES for key encryption (3DES disallowed)
# - Use KeyStore.getDefaultType() instead of hardcoded "JKS" or "PKCS12"
# - Remove explicit BouncyCastleProvider registration (use system-configured provider)
# NOTE: CURRENTLY UNUSED IN BUILD BECAUSE OF SCRAM AND PEM TESTS
diff --git a/clients/src/test/java/org/apache/kafka/test/TestSslUtils.java b/clients/src/test/java/org/apache/kafka/test/TestSslUtils.java
index b89aade..0d27e08 100644
--- a/clients/src/test/java/org/apache/kafka/test/TestSslUtils.java
+++ b/clients/src/test/java/org/apache/kafka/test/TestSslUtils.java
@@ -31,7 +31,6 @@ import org.bouncycastle.asn1.x500.AttributeTypeAndValue;
 import org.bouncycastle.asn1.x500.RDN;
 import org.bouncycastle.asn1.x500.X500Name;
 import org.bouncycastle.asn1.x500.style.BCStyle;
-import org.bouncycastle.asn1.x509.AlgorithmIdentifier;
 import org.bouncycastle.asn1.x509.BasicConstraints;
 import org.bouncycastle.asn1.x509.Extension;
 import org.bouncycastle.asn1.x509.GeneralName;
@@ -41,20 +40,12 @@ import org.bouncycastle.asn1.x509.SubjectPublicKeyInfo;
 import org.bouncycastle.cert.X509CertificateHolder;
 import org.bouncycastle.cert.X509v3CertificateBuilder;
 import org.bouncycastle.cert.jcajce.JcaX509CertificateConverter;
-import org.bouncycastle.crypto.params.AsymmetricKeyParameter;
-import org.bouncycastle.crypto.util.PrivateKeyFactory;
-import org.bouncycastle.jce.provider.BouncyCastleProvider;
 import org.bouncycastle.openssl.PKCS8Generator;
 import org.bouncycastle.openssl.jcajce.JcaMiscPEMGenerator;
 import org.bouncycastle.openssl.jcajce.JcaPKCS8Generator;
 import org.bouncycastle.openssl.jcajce.JceOpenSSLPKCS8EncryptorBuilder;
 import org.bouncycastle.operator.ContentSigner;
-import org.bouncycastle.operator.DefaultDigestAlgorithmIdentifierFinder;
-import org.bouncycastle.operator.DefaultSignatureAlgorithmIdentifierFinder;
-import org.bouncycastle.operator.bc.BcContentSignerBuilder;
-import org.bouncycastle.operator.bc.BcDSAContentSignerBuilder;
-import org.bouncycastle.operator.bc.BcECContentSignerBuilder;
-import org.bouncycastle.operator.bc.BcRSAContentSignerBuilder;
+import org.bouncycastle.operator.jcajce.JcaContentSignerBuilder;
 import org.bouncycastle.util.io.pem.PemWriter;
 
 import java.io.ByteArrayOutputStream;
@@ -79,7 +70,6 @@ import java.security.KeyStore;
 import java.security.NoSuchAlgorithmException;
 import java.security.PrivateKey;
 import java.security.SecureRandom;
-import java.security.Security;
 import java.security.cert.Certificate;
 import java.security.cert.CertificateException;
 import java.security.cert.X509Certificate;
@@ -158,7 +148,7 @@ public class TestSslUtils {
     }
 
     private static KeyStore createEmptyKeyStore() throws GeneralSecurityException, IOException {
-        KeyStore ks = KeyStore.getInstance("JKS");
+        KeyStore ks = KeyStore.getInstance(KeyStore.getDefaultType());
         ks.load(null, null); // initialize
         return ks;
     }
@@ -193,7 +183,7 @@ public class TestSslUtils {
 
     public static <T extends Certificate> void createTrustStore(
             String filename, Password password, Map<String, T> certs) throws GeneralSecurityException, IOException {
-        KeyStore ks = KeyStore.getInstance("JKS");
+        KeyStore ks = KeyStore.getInstance(KeyStore.getDefaultType());
         try (InputStream in = Files.newInputStream(Paths.get(filename))) {
             ks.load(in, password.value().toCharArray());
         } catch (EOFException e) {
@@ -381,7 +371,7 @@ public class TestSslUtils {
             if (password == null) {
                 pemWriter.writeObject(new JcaPKCS8Generator(privateKey, null));
             } else {
-                JceOpenSSLPKCS8EncryptorBuilder encryptorBuilder = new JceOpenSSLPKCS8EncryptorBuilder(PKCS8Generator.PBE_SHA1_3DES);
+                JceOpenSSLPKCS8EncryptorBuilder encryptorBuilder = new JceOpenSSLPKCS8EncryptorBuilder(PKCS8Generator.AES_128_CBC);
                 encryptorBuilder.setPassword(password.value().toCharArray());
                 try {
                     pemWriter.writeObject(new JcaPKCS8Generator(privateKey, encryptorBuilder.build()));
@@ -399,7 +389,7 @@ public class TestSslUtils {
         private byte[] subjectAltName;
 
         public CertificateBuilder() {
-            this(30, "SHA1withRSA");
+            this(30, "SHA256withRSA");
         }
 
         public CertificateBuilder(int days, String algorithm) {
@@ -437,22 +427,8 @@ public class TestSslUtils {
 
         public X509Certificate generate(X500Name dn, KeyPair keyPair) throws CertificateException {
             try {
-                Security.addProvider(new BouncyCastleProvider());
-                AlgorithmIdentifier sigAlgId = new DefaultSignatureAlgorithmIdentifierFinder().find(algorithm);
-                AlgorithmIdentifier digAlgId = new DefaultDigestAlgorithmIdentifierFinder().find(sigAlgId);
-                AsymmetricKeyParameter privateKeyAsymKeyParam = PrivateKeyFactory.createKey(keyPair.getPrivate().getEncoded());
                 SubjectPublicKeyInfo subPubKeyInfo = SubjectPublicKeyInfo.getInstance(keyPair.getPublic().getEncoded());
-                BcContentSignerBuilder signerBuilder;
-                String keyAlgorithm = keyPair.getPublic().getAlgorithm();
-                if (keyAlgorithm.equals("RSA"))
-                    signerBuilder = new BcRSAContentSignerBuilder(sigAlgId, digAlgId);
-                else if (keyAlgorithm.equals("DSA"))
-                    signerBuilder = new BcDSAContentSignerBuilder(sigAlgId, digAlgId);
-                else if (keyAlgorithm.equals("EC"))
-                    signerBuilder = new BcECContentSignerBuilder(sigAlgId, digAlgId);
-                else
-                    throw new IllegalArgumentException("Unsupported algorithm " + keyAlgorithm);
-                ContentSigner sigGen = signerBuilder.build(privateKeyAsymKeyParam);
+                ContentSigner sigGen = new JcaContentSignerBuilder(algorithm).build(keyPair.getPrivate());
                 // Negative numbers for "days" can be used to generate expired certificates
                 Date now = new Date();
                 Date from = (days >= 0) ? now : new Date(now.getTime() + days * 86400000L);
@@ -463,7 +439,7 @@ public class TestSslUtils {
                 if (subjectAltName != null)
                     v3CertGen.addExtension(Extension.subjectAlternativeName, false, subjectAltName);
                 X509CertificateHolder certificateHolder = v3CertGen.build(sigGen);
-                return new JcaX509CertificateConverter().setProvider("BC").getCertificate(certificateHolder);
+                return new JcaX509CertificateConverter().getCertificate(certificateHolder);
             } catch (CertificateException ce) {
                 throw ce;
             } catch (Exception e) {
@@ -506,29 +482,12 @@ public class TestSslUtils {
                 int daysBeforeNow, int daysAfterNow, X500Name issuer, KeyPair parentKeyPair, boolean isCA, boolean isServerCert, boolean isClientCert)
                 throws CertificateException {
             try {
-                Security.addProvider(new BouncyCastleProvider());
-                AlgorithmIdentifier sigAlgId =
-                        new DefaultSignatureAlgorithmIdentifierFinder().find(algorithm);
-                AlgorithmIdentifier digAlgId =
-                        new DefaultDigestAlgorithmIdentifierFinder().find(sigAlgId);
                 // Create self-signed certificate if no parentKeyPair has been specified, otherwise
                 // sign with private key of parentKeyPair
                 KeyPair signingKeyPair = (parentKeyPair != null) ? parentKeyPair : keyPair;
-                AsymmetricKeyParameter privateKeyAsymKeyParam =
-                        PrivateKeyFactory.createKey(signingKeyPair.getPrivate().getEncoded());
                 SubjectPublicKeyInfo subPubKeyInfo =
                         SubjectPublicKeyInfo.getInstance(keyPair.getPublic().getEncoded());
-                BcContentSignerBuilder signerBuilder;
-                String keyAlgorithm = keyPair.getPublic().getAlgorithm();
-                if (keyAlgorithm.equals("RSA"))
-                    signerBuilder = new BcRSAContentSignerBuilder(sigAlgId, digAlgId);
-                else if (keyAlgorithm.equals("DSA"))
-                    signerBuilder = new BcDSAContentSignerBuilder(sigAlgId, digAlgId);
-                else if (keyAlgorithm.equals("EC"))
-                    signerBuilder = new BcECContentSignerBuilder(sigAlgId, digAlgId);
-                else
-                    throw new IllegalArgumentException("Unsupported algorithm " + keyAlgorithm);
-                ContentSigner sigGen = signerBuilder.build(privateKeyAsymKeyParam);
+                ContentSigner sigGen = new JcaContentSignerBuilder(algorithm).build(signingKeyPair.getPrivate());
                 // Negative numbers for "days" can be used to generate expired certificates
                 Date now = new Date();
                 Date from = new Date(now.getTime() - daysBeforeNow * 86400000L);
@@ -554,8 +513,7 @@ public class TestSslUtils {
                     v3CertGen.addExtension(Extension.subjectAlternativeName, false, subjectAltName);
                 }
                 X509CertificateHolder certificateHolder = v3CertGen.build(sigGen);
-                return new JcaX509CertificateConverter().setProvider("BC")
-                        .getCertificate(certificateHolder);
+                return new JcaX509CertificateConverter().getCertificate(certificateHolder);
             } catch (CertificateException ce) {
                 throw ce;
             } catch (Exception e) {
@@ -676,7 +634,7 @@ public class TestSslUtils {
 
             if (connectionMode == ConnectionMode.SERVER || (connectionMode == ConnectionMode.CLIENT && keyStoreFile != null)) {
                 sslConfigs.put(SslConfigs.SSL_KEYSTORE_LOCATION_CONFIG, keyStoreFile.getPath());
-                sslConfigs.put(SslConfigs.SSL_KEYSTORE_TYPE_CONFIG, "JKS");
+                sslConfigs.put(SslConfigs.SSL_KEYSTORE_TYPE_CONFIG, KeyStore.getDefaultType());
                 sslConfigs.put(SslConfigs.SSL_KEYMANAGER_ALGORITHM_CONFIG, TrustManagerFactory.getDefaultAlgorithm());
                 sslConfigs.put(SslConfigs.SSL_KEYSTORE_PASSWORD_CONFIG, keyStorePassword);
                 sslConfigs.put(SslConfigs.SSL_KEY_PASSWORD_CONFIG, keyPassword);
@@ -684,7 +642,7 @@ public class TestSslUtils {
 
             sslConfigs.put(SslConfigs.SSL_TRUSTSTORE_LOCATION_CONFIG, trustStoreFile.getPath());
             sslConfigs.put(SslConfigs.SSL_TRUSTSTORE_PASSWORD_CONFIG, trustStorePassword);
-            sslConfigs.put(SslConfigs.SSL_TRUSTSTORE_TYPE_CONFIG, "JKS");
+            sslConfigs.put(SslConfigs.SSL_TRUSTSTORE_TYPE_CONFIG, KeyStore.getDefaultType());
             sslConfigs.put(SslConfigs.SSL_TRUSTMANAGER_ALGORITHM_CONFIG, TrustManagerFactory.getDefaultAlgorithm());
 
             List<String> enabledProtocols  = new ArrayList<>();
@@ -807,14 +765,14 @@ public class TestSslUtils {
 
         File keystoreStoreFile = TestUtils.tempFile("keystore", ".jks");
         Password keyStorePassword =  new Password("password");
-        KeyStore keyStore = KeyStore.getInstance("PKCS12");
+        KeyStore keyStore = KeyStore.getInstance(KeyStore.getDefaultType());
         keyStore.load(null, null);
         keyStore.setKeyEntry("issued-cert", keyPairs[0].getPrivate(), keyStorePassword.value().toCharArray(), certs);
         saveKeyStore(keyStore, keystoreStoreFile.getPath(), keyStorePassword);
 
         File trustStoreFile = TestUtils.tempFile("truststore", ".jks");
         Password trustStorePassword =  new Password("password");
-        KeyStore trustStore = KeyStore.getInstance("PKCS12");
+        KeyStore trustStore = KeyStore.getInstance(KeyStore.getDefaultType());
         trustStore.load(null, null);
         for (X509Certificate cert : certs) {
             trustStore.setCertificateEntry(cert.getSubjectX500Principal().getName(), cert);
@@ -825,14 +783,14 @@ public class TestSslUtils {
 
         sslConfigs.put(SslConfigs.SSL_PROTOCOL_CONFIG, tlsProtocol); // protocol to create SSLContext
         sslConfigs.put(SslConfigs.SSL_KEYSTORE_LOCATION_CONFIG, keystoreStoreFile.getPath());
-        sslConfigs.put(SslConfigs.SSL_KEYSTORE_TYPE_CONFIG, "JKS");
+        sslConfigs.put(SslConfigs.SSL_KEYSTORE_TYPE_CONFIG, KeyStore.getDefaultType());
         sslConfigs.put(SslConfigs.SSL_KEYMANAGER_ALGORITHM_CONFIG, TrustManagerFactory.getDefaultAlgorithm());
         sslConfigs.put(SslConfigs.SSL_KEYSTORE_PASSWORD_CONFIG, keyStorePassword);
         sslConfigs.put(SslConfigs.SSL_KEY_PASSWORD_CONFIG, keyStorePassword);
 
         sslConfigs.put(SslConfigs.SSL_TRUSTSTORE_LOCATION_CONFIG, trustStoreFile.getPath());
         sslConfigs.put(SslConfigs.SSL_TRUSTSTORE_PASSWORD_CONFIG, trustStorePassword);
-        sslConfigs.put(SslConfigs.SSL_TRUSTSTORE_TYPE_CONFIG, "JKS");
+        sslConfigs.put(SslConfigs.SSL_TRUSTSTORE_TYPE_CONFIG, KeyStore.getDefaultType());
         sslConfigs.put(SslConfigs.SSL_TRUSTMANAGER_ALGORITHM_CONFIG, TrustManagerFactory.getDefaultAlgorithm());
 
         List<String> enabledProtocols  = new ArrayList<>();
